# Ujian Akhir Semester (6)

Nama   : Clarence Chandra Sentosa

NRP    : 1973010

# Training Classifier dengan KNN (K Nearest Neighbour)

# Load Breast Cancer Dataset

In [18]:
# Code 01 (Import Modul)

from sklearn.datasets import load_breast_cancer

cancer = load_breast_cancer()

# Dataset Dimension

In [20]:
# Code 02 (Menampilkan dimensi dataset)

cancer['data'].shape

(569, 30)

Kita bisa mengetahui pada code 02 bahwa **Breast Cancer** dataset memiliki 569 baris dan 30 kolom.

# Training Set dan Testing Set

Kita akan membagi dataset kedalam *Training Set* dan *Testing Set* nya.

In [25]:
# Code 03

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(
    cancer['data'], cancer['target'], random_state = 0)

print("X_train shape", format(X_train.shape))
print("Y_train shape", format(y_train.shape))

print("X_test shape", format(X_test.shape))
print("Y_test shape", format(y_test.shape))

X_train shape (426, 30)
Y_train shape (426,)
X_test shape (143, 30)
Y_test shape (143,)


**Konsep berpikir code 03 :**
- Pertama-tama kita import terlebih dahulu modulnya dengan memanggil *from sklearn.model_selection import train_test_split*.
- Lalu kita panggil *train_test_split* dengan menyertakan parameter *(cancer['data'], cancer['target'], random_state = 0)*.
- Lalu kita tampikan *X_train* dan *X_test* nya untuk melihat dimensi dari training set dan testing set nya.

Berdasarkan hasil output code 03, kita bisa lihat bahwa X_train shape (426, 30) dan X_test shape (143, 30).
    


# Training KNN Classification Model 

Setelah training setnya kita persiapkan, selnjutnya kita akan melangkah ke proses training model. Kali ini model machine learning yang akan kita gunakan adalah **KNN**.

In [32]:
# Code 04

from sklearn.neighbors import KNeighborsClassifier

knn = KNeighborsClassifier(n_neighbors = 1)

knn.fit(X_train, y_train)

KNeighborsClassifier(n_neighbors=1)

**Konsep berpikir code 04 :**
- Pertama-tama kita akan import terlebih dahulu estimater class nya 'from sklearn.neighbors import KNeighborsClassifier'.
- Lalu kita panggil *KNeighborsClassifier* dengan menyertakan parameter *(n_neighbors = 1)* artinya kita menentukan jumlah tetangga pada kasus ini ialah 1 tetangga terdekat yang kita tampung kedalam variabel *knn*.
- Selanjutnya kita akan melakukan fit transfrom pada *X_train* dan *y_train* dengan memanggil *fit(X_train, y_train)* yang kita kenakan pada variabel *knn*. 

# Prediction Breast Cancer Dataset by KNN 

Selanjutnya kita akan melakukan prediksi terhadap *X_test* yang dimiliki.

In [34]:
# Code 05 (Prediction Dataset)

y_pred = knn.predict(X_test)

# Classification Report

In [36]:
# Code 06

from sklearn.metrics import classification_report

cls_report = classification_report(y_test, y_pred)

print(f'Classification Report :\n{cls_report}')

Classification Report :
              precision    recall  f1-score   support

           0       0.90      0.87      0.88        53
           1       0.92      0.94      0.93        90

    accuracy                           0.92       143
   macro avg       0.91      0.91      0.91       143
weighted avg       0.92      0.92      0.92       143



**Alur berpikir code 06** :
- Pertama-tam kita akan import terlebih dahulu modulnya dengan memanggil *'from sklearn.metrics import classification_report'*.
- Untuk selanjutnya kita akan memanggil *'classification_report'* dengan menyertakan parameter *'(y_test, y_pred)'* yang kita tampung kedalam variabel *cls_report*.

Hasil output code 06 merupakan *classification report* sebagai bentuk evaluasi terhadap model **KNN** pada **Breast Cancer** dataset yang baru saja kita training sebelumnya, ditampilkan nilai *precision*, *recall*, serta *f1-score* nya.

Demikianlah cara yang saya lakukan untuk melakukan training classifier dengan menggunakan *Key Nearest Neighbour* atau **KNN** pada **Breast Cancer** dataset.

# "Banyak Belajar Biar Bisa Bantu Banyak Orang"

Written by : Clarence Code Pianist (1973010)